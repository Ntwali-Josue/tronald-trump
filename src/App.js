import './styles/App.css';
import Quotes from './components/Quotes';

const App = () => (
  <div className="App">
    <Quotes />
  </div>
);

export default App;
