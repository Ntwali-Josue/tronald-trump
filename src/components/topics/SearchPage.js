import React from 'react';
import { NavLink } from 'react-router-dom';
import { useDispatch, useSelector } from 'react-redux';
import Topic from './Topic';
import { fetchQuotes } from '../../redux/reducers/quotes';

const SearchPage = () => {
  const dispatch = useDispatch();
  const handlePickTopic = (topic) => {
    dispatch(fetchQuotes(topic));
  };
  const topic = useSelector((state) => state.quotes.filteredTopics);
  return (
    <div className="details-container">
      <NavLink className="back-btn" to="/">
        <img alt="Back button" src="https://img.icons8.com/ios-glyphs/30/ffffff/circled-left-2.png" />
      </NavLink>
      <div className="filtered-topic">
        <NavLink className="text-white" key={topic.id} to="/details" onClick={() => handlePickTopic(topic.name)}>
          <Topic topicName={topic.name} image={topic.images} count={topic.count} />
        </NavLink>
      </div>
    </div>
  );
};
export default SearchPage;
