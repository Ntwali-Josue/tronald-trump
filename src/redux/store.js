import { createStore, combineReducers, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import logger from 'redux-logger';
import quotesReducer from './reducers/quotes';

const reducer = combineReducers({
  quotes: quotesReducer,
});

const store = createStore(
  reducer,
  applyMiddleware(thunk, logger),
);

export default store;
