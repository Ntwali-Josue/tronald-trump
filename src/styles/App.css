.App {
  margin: 0;
}

.details-container {
  background-color: #5689e5;
  height: 85vh;
  color: white;
  text-align: center;
  font-size: 25px;
  display: flex;
  flex-direction: column;
  align-items: center;
  row-gap: 10px;
  padding-top: 15px;
}

.back-btn {
  width: 100%;
  display: flex;
  justify-content: start;
  padding-left: 10%;
}
