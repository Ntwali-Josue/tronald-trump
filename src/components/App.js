import { BrowserRouter as Routers, Routes, Route } from 'react-router-dom';
import '../styles/App.css';
import Home from './topics/TopicsPage';
import Quotes from './quotes/Quotes';
import NavBar from './NavBar';
import SearchPage from './topics/SearchPage';

const App = () => (
  <div>
    <Routers>
      <NavBar />
      <Routes>
        <Route path="/" element={<Home />} />
        <Route path="/details" element={<Quotes />} />
        <Route path="/search" element={<SearchPage />} />
      </Routes>
    </Routers>
  </div>
);

export default App;
