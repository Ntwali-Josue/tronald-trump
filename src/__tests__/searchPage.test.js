import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import SearchPage from '../components/topics/SearchPage';
import App from '../components/App';
import store from '../redux/store';

describe('pages test snapshots', () => {
  it('testing search Page', async () => {
    const searchPage = renderer
      .create(
        <Provider store={store}>
          <App>
            <SearchPage />
          </App>
        </Provider>,
      )
      .toJSON();
    expect(searchPage).toMatchSnapshot();
  });
});
