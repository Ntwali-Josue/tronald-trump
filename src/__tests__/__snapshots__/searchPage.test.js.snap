// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pages test snapshots testing topics Page 1`] = `
<div>
  <header
    className="nav-header"
  >
    <img
      alt="Mobile menu button"
      src="https://img.icons8.com/material-rounded/24/ffffff/menu--v1.png"
    />
    <a
      aria-current="page"
      className="active"
      href="/"
      onClick={[Function]}
    >
      <h1
        className="nav-title"
      >
        Trump Qoutes
      </h1>
    </a>
    <div
      className="nav-icon"
    >
      <img
        alt="config button"
        src="https://img.icons8.com/external-those-icons-fill-those-icons/24/ffffff/external-gear-settings-those-icons-fill-those-icons-1.png"
      />
    </div>
  </header>
  <div
    className="topics-container"
  >
    <div
      className="search"
    >
      <span>
        Search by topic name
      </span>
      <select
        id="topics"
        name="topics"
        onChange={[Function]}
      >
        <option
          placeholder="topic"
          value="WOMEN"
        >
          WOMEN
        </option>
        <option
          placeholder="topic"
          value="OBAMA"
        >
          OBAMA
        </option>
        <option
          placeholder="topic"
          value="MONEY"
        >
          MONEY
        </option>
        <option
          placeholder="topic"
          value="TRUMP"
        >
          TRUMP
        </option>
        <option
          placeholder="topic"
          value="CLINTON"
        >
          CLINTON
        </option>
        <option
          placeholder="topic"
          value="MUSLIMS"
        >
          MUSLIMS
        </option>
        <option
          placeholder="topic"
          value="HISTORY"
        >
          HISTORY
        </option>
        <option
          placeholder="topic"
          value="TED CRUZ"
        >
          TED CRUZ
        </option>
      </select>
      <input
        type="submit"
        value="Search"
      />
    </div>
    <ul
      className="topic-section"
    >
      <a
        className="topic-route"
        href="/details"
        onClick={[Function]}
      >
        <li
          className="topic-item"
        >
          <div>
            <img
              alt="images"
              className="image"
              src="https://icons-for-free.com/iconfiles/png/512/female+person+user+woman+young+icon-1320196266256009072.png"
            />
            <h1
              className="topic-title"
            >
              WOMEN
              <img
                alt="Front button"
                src="https://img.icons8.com/ios-glyphs/30/ffffff/circled-right-2.png"
              />
            </h1>
            <span
              className="qoutes-number"
            >
              10 Qoutes
            </span>
          </div>
        </li>
      </a>
      <a
        className="topic-route"
        href="/details"
        onClick={[Function]}
      >
        <li
          className="topic-item"
        >
          <div>
            <img
              alt="images"
              className="image"
              src="https://purepng.com/public/uploads/large/purepng.com-barack-obamabarack-obamapoliticianamericanpresidentserved-1701528015349anvkm.png"
            />
            <h1
              className="topic-title"
            >
              OBAMA
              <img
                alt="Front button"
                src="https://img.icons8.com/ios-glyphs/30/ffffff/circled-right-2.png"
              />
            </h1>
            <span
              className="qoutes-number"
            >
              10 Qoutes
            </span>
          </div>
        </li>
      </a>
      <a
        className="topic-route"
        href="/details"
        onClick={[Function]}
      >
        <li
          className="topic-item"
        >
          <div>
            <img
              alt="images"
              className="image"
              src="https://cdn3.iconfinder.com/data/icons/glypho-money-and-finance/64/money-dollar-circle-512.png"
            />
            <h1
              className="topic-title"
            >
              MONEY
              <img
                alt="Front button"
                src="https://img.icons8.com/ios-glyphs/30/ffffff/circled-right-2.png"
              />
            </h1>
            <span
              className="qoutes-number"
            >
              10 Qoutes
            </span>
          </div>
        </li>
      </a>
      <a
        className="topic-route"
        href="/details"
        onClick={[Function]}
      >
        <li
          className="topic-item"
        >
          <div>
            <img
              alt="images"
              className="image"
              src="https://cdn0.iconfinder.com/data/icons/politicians-vol-2/479/trump_grey-512.png"
            />
            <h1
              className="topic-title"
            >
              TRUMP
              <img
                alt="Front button"
                src="https://img.icons8.com/ios-glyphs/30/ffffff/circled-right-2.png"
              />
            </h1>
            <span
              className="qoutes-number"
            >
              10 Qoutes
            </span>
          </div>
        </li>
      </a>
      <a
        className="topic-route"
        href="/details"
        onClick={[Function]}
      >
        <li
          className="topic-item"
        >
          <div>
            <img
              alt="images"
              className="image"
              src="https://upload.wikimedia.org/wikipedia/commons/thumb/6/66/Clinton_Circle.png/800px-Clinton_Circle.png"
            />
            <h1
              className="topic-title"
            >
              CLINTON
              <img
                alt="Front button"
                src="https://img.icons8.com/ios-glyphs/30/ffffff/circled-right-2.png"
              />
            </h1>
            <span
              className="qoutes-number"
            >
              10 Qoutes
            </span>
          </div>
        </li>
      </a>
      <a
        className="topic-route"
        href="/details"
        onClick={[Function]}
      >
        <li
          className="topic-item"
        >
          <div>
            <img
              alt="images"
              className="image"
              src="https://cdn-icons-png.flaticon.com/512/190/190600.png"
            />
            <h1
              className="topic-title"
            >
              MUSLIMS
              <img
                alt="Front button"
                src="https://img.icons8.com/ios-glyphs/30/ffffff/circled-right-2.png"
              />
            </h1>
            <span
              className="qoutes-number"
            >
              2 Qoutes
            </span>
          </div>
        </li>
      </a>
      <a
        className="topic-route"
        href="/details"
        onClick={[Function]}
      >
        <li
          className="topic-item"
        >
          <div>
            <img
              alt="images"
              className="image"
              src="https://previews.123rf.com/images/vectorstockcompany/vectorstockcompany1809/vectorstockcompany180900008/108986964-history-icon-linear-simple-element-illustration-clock-with-arrow-concept-outline-symbol-design-vecto.jpg"
            />
            <h1
              className="topic-title"
            >
              HISTORY
              <img
                alt="Front button"
                src="https://img.icons8.com/ios-glyphs/30/ffffff/circled-right-2.png"
              />
            </h1>
            <span
              className="qoutes-number"
            >
              10 Qoutes
            </span>
          </div>
        </li>
      </a>
      <a
        className="topic-route"
        href="/details"
        onClick={[Function]}
      >
        <li
          className="topic-item"
        >
          <div>
            <img
              alt="images"
              className="image"
              src="https://www.gannett-cdn.com/-mm-/2495659f3130e7ec633839597cc03a63838d0a2c/c=0-320-3840-5440/local/-/media/2017/06/22/TXGroup/Abilene/636337541178608717-Ted-Cruz.jpg?width=660&height=880&fit=crop&format=pjpg&auto=webp"
            />
            <h1
              className="topic-title"
            >
              TED CRUZ
              <img
                alt="Front button"
                src="https://img.icons8.com/ios-glyphs/30/ffffff/circled-right-2.png"
              />
            </h1>
            <span
              className="qoutes-number"
            >
              10 Qoutes
            </span>
          </div>
        </li>
      </a>
    </ul>
  </div>
</div>
`;
